File: service-common/src/main/java/me/reidj/bridgebuilders/packages/TopPackage.java
package me.reidj.bridgebuilders.packages;

import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.RequiredArgsConstructor;
import me.reidj.bridgebuilders.tops.PlayerTopEntry;
import ru.cristalix.core.network.CorePackage;

import java.util.List;

@Data
@EqualsAndHashCode(callSuper = false)
@RequiredArgsConstructor
public class TopPackage extends CorePackage {

    // request
    private final TopType topType;
    private final int limit;

    // response
    private List<PlayerTopEntry<Object>> entries;

    public enum TopType {

        WINS,
        GAMES
        ;

    }

}

File: commons/src/main/kotlin/me/reidj/bridgebuilders/package/TopPackage.kt
package me.reidj.bridgebuilders.`package`

import me.reidj.bridgebuilders.top.PlayerTopEntry

data class TopPackage(val type: TopType, val limit: Int): BridgePackage() {
    lateinit var entries: List<PlayerTopEntry<Any>>

    enum class TopType {
        WINS,
        ;
    }
}
